pkg/services/librarypanels/api.go

```diff
@@ -49,3 +51,28 @@ func (lps *LibraryPanelService) deleteHandler(c *models.ReqContext) api.Response
 
 	return api.Success("Library panel deleted")
 }
+
+// getHandler handles GET /api/library-panels/:panelId
+func (lps *LibraryPanelService) getHandler(c *models.ReqContext) api.Response {
+	libraryPanel, err := lps.getLibraryPanel(c, c.ParamsInt64(":panelId"))
+
+	if err != nil {
+		if errors.Is(err, errLibraryPanelNotFound) {
+			return api.Error(404, errLibraryPanelNotFound.Error(), err)
+		}
+		return api.Error(500, "Failed to get library panel", err)
+	}
+
+	return api.JSON(200, util.DynMap{"result": libraryPanel})
+}
+
+// getAllHandler handles GET /api/library-panels/
+func (lps *LibraryPanelService) getAllHandler(c *models.ReqContext) api.Response {
```

Comment:
think this will need some search, page  & page size at some point, but good to start simple :)

Reply:
Yes, keeping it simple for now :)